stages/org.osbuild.systemd.unit.create

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/python3
+"""
+Create Systemd services unit file
+
+This stage allows to create Systemd unit files in
+`/usr/lib/systemd/system/`. The `filename` property specifies the
+'.service' file to be added. These names are validated using the
+same rules as specified by systemd.unit(5) and they must contain the
+'.service' suffix (other types of unit files are not supported).
+
+The Unit configuration can currently specify the following subset
+of options:
+  - 'Unit' section
+    - <string> options
+  - 'Service' section
+    - <string> options
+  - 'Install' section
+    - <string> options
+"""
+
+import configparser
+import sys
+
+import osbuild.api
+
+SCHEMA = r"""
+"additionalProperties": false,
+"required": ["filename","config"],
+"properties": {
+  "filename": {
+    "type": "string",
+    "pattern": "^[\\w:.\\\\-]+[@]{0,1}[\\w:.\\\\-]*\\.service$"
+  },
+  "config": {
+    "additionalProperties": false,
+    "type": "object",
+    "description": "Configuration for a '.service' unit.",
+    "properties": {
+      "Unit": {
+        "additionalProperties": false,
+        "type": "array",
+        "description": "'Unit' configuration section of a unit file.",
+        "items": {
+          "type": "string",
+          "description": "Parameters for unit section"
+        }
+      },
+      "Service": {
+        "additionalProperties": false,
+        "type": "array",
+        "description": "'Service' configuration section of a unit file.",
+        "items": {
+          "type": "string",
+          "description": "Parameters for Service section"
+        }
+      },
+      "Install": {
+        "additionalProperties": false,
+        "type": "array",
+        "description": "'Install' configuration section of a unit file.",
+        "items": {
+          "type": "string",
+          "description": "Parameters for install section"
+        }
+      }
+    }
+  }
+}
+"""
+
+
+def main(tree, options):
+    filename = options["filename"]
+    cfg = options["config"]
+    # ensure the service name does not exceed maximum filename length
+    if len(filename) > 255:
+        raise ValueError(f"Error: the {filename} service exceeds the maximum filename length.")
+
+    config = configparser.ConfigParser()
+    # prevent conversion of the option name to lowercase
+    config.optionxform = lambda option: option
+
+    for section, opts in cfg.items():
+        if not config.has_section(section):
+            config.add_section(section)
+        for opt in opts:
+            key_value_pair = str(opt).split("=", 1)  # maxsplit set to split 1st occurance of '='
+            if len(key_value_pair) == 1:
+                return 1
+            config.set(section, key_value_pair[0], key_value_pair[1])
```

Comment:
Just a random thought: systemd allows specifying assignments multiple times, but configparser doesn't support it. If we want to expose this option, we might need to use something else than configparser. See https://bugs.python.org/issue31528 and

```
[Service]  
Type=oneshot
ExecStart=mkfs.ext4 /dev/sdb
ExecStart=mkdir /mnt/data
ExecStart=mount /dev/sdb /mnt/data
ExecStart=bash -c "echo '/dev/sdb /mnt/data ext4 defaults 0 2' >>/etc/fstab"
ExecStartPost=touch /var/lib/prepare-data-disk-first-boot
```

Reply:
Nice catch!

Reply:
Quick solution was to allow_now_value in config parser and put the whole string as option without value.
I think we wont have duplicate systemd option:value combo under a section.